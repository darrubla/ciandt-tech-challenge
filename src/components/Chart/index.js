/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import PropTypes from 'prop-types'

import RadarChart from 'react-svg-radar-chart'
import 'react-svg-radar-chart/build/css/index.css'

function Chart({ data, captions }) {
  const props = {
    captionProps: () => ({
      className: 'caption',
      textAnchor: 'middle',
      fontSize: 20,
      fontFamily: 'sans-serif',
    }),
  }
  return <RadarChart captions={captions} data={data} size={250} {...props} />
}

Chart.propTypes = {
  data: PropTypes.object.isRequired,
  captions: PropTypes.object.isRequired,
}

export default Chart
