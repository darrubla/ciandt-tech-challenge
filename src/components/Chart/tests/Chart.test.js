import React from 'react'
import { render } from '@testing-library/react'

import Chart from '../index'

describe('<Chart />', () => {
  it('Should render properly', () => {
    const props = {
      captions: {
        test: 'test'
      },
      data: [{
        data: { test: 0.2 },
        meta: {
          color: 'red'
        }
      }],
    }
    const { getByText } = render(<Chart {...props} />)
    expect(getByText('test')).toBeTruthy()
  })

  it('Should render and match the snapshot', () => {
    const props = {
      captions: {
        test: 'test'
      },
      data: [{
        data: { test: 0.2 },
        meta: {
          color: 'red'
        }
      }],
    }
    const {
      container: { firstChild },
    } = render(<Chart {...props} />)
    expect(firstChild).toMatchSnapshot()
  })
})