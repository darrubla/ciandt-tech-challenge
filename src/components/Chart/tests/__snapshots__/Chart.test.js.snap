// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart /> Should render and match the snapshot 1`] = `
<svg
  height="250"
  version="1"
  viewBox="-10 0 270 250"
  width="250"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    transform="translate(125.0000,125.0000)"
  >
    <g>
      <circle
        class="scale"
        cx="0"
        cy="0"
        fill="none"
        r="104.16666666666667"
      />
      <circle
        class="scale"
        cx="0"
        cy="0"
        fill="none"
        r="69.44444444444444"
      />
      <circle
        class="scale"
        cx="0"
        cy="0"
        fill="none"
        r="34.72222222222222"
      />
    </g>
    <g>
      <polyline
        class="axis"
        points="0.0000,0.0000 0.0000,-104.1667"
      />
    </g>
    <g>
      <path
        class="shape "
        d="M0.0000,-20.8333z"
        fill="red"
        stroke="red"
      />
    </g>
    <g>
      <text
        class="caption"
        dy="5"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        x="0.0000"
        y="-118.7500"
      >
        test
      </text>
    </g>
  </g>
</svg>
`;
